### PR TITLE
[DependencyInjection][Framework] `kernel.locale_aware`

### PR DESCRIPTION
**Pull request by Guikingone**
Hi 👋🏻 

It seems that the `kernel.locale_aware` tag was introduced in `4.3` without documenting it, here's a first draft.

Thanks for the feedback 🙂

### CODE REVIEW

File: reference/dic_tags.rst [hunk 1]
```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
 :class:`Symfony\\Component\\HttpKernel\\Fragment\\FragmentRendererInterface`,
 register it as a service, then tag it with ``kernel.fragment_renderer``.
 
+kernel.locale_aware
+-------------------
+
+.. versionadded:: 4.3
```
Comment (94noni) [inline reference/dic_tags.rst:459]: You need to target v4.4 and here swap to 4.4 :)

File: reference/dic_tags.rst [hunk 2]
```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
...
+
+    The ``kernel.locale_aware`` tag was introduced in Symfony 4.3.
+
+**Purpose**: To interact with the locale
```
Comment (94noni) [inline reference/dic_tags.rst]: Perhaps add link:ref to what is a locale in symfony request/response context?

File: reference/dic_tags.rst [hunk 3]
```diff
@@ -26,6 +26,7 @@ Tag Name                                  Usage
 `kernel.event_listener`_                  Listen to different events/hooks in Symfony
 `kernel.event_subscriber`_                To subscribe to a set of different events/hooks in Symfony
 `kernel.fragment_renderer`_               Add new HTTP content rendering strategies
+`kernel.locale_aware`_                    To interact with the locale
```
Comment (HeahDude) [inline reference/dic_tags.rst]:
```suggestion
`kernel.locale_aware`_                    To access and use the current locale
```
?

File: reference/dic_tags.rst (hunk 2 above)
Comment (HeahDude) [reply reference/dic_tags.rst]: I would remove this line

File: reference/dic_tags.rst [hunk 4]
```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
...
+-------------------
+
+.. versionadded:: 4.3
+
+    The ``kernel.locale_aware`` tag was introduced in Symfony 4.3.
+
+**Purpose**: To interact with the locale
+
+Setting and retrieving the locale can be done via the configuration or
+thanks to container parameters or even listeners / route parameters.
```
Comment (HeahDude) [inline reference/dic_tags.rst]:
```suggestion
using container parameters, even listeners, route parameters or the current request.
```
?

File: reference/dic_tags.rst [hunk 5]
```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
...
+    // src/Locale/MyCustomLocaleHandler.php
+    namespace App\Locale;
+
+    use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+    class MyCustomLocaleHandler implements LocaleAwareInterface
+    {
+        public function setLocale($locale)
+        {
+            // ... do some sort of operations to set the locale
```
Comment (HeahDude) [inline reference/dic_tags.rst]: This setter should not be called as it used to pass the locale to the context of the class when built or when using the new switcher.
I would remove this comment and hard code a proper assignation instead:
```php
$this->locale = $locale;
```

File: reference/dic_tags.rst [hunk 6]
```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
...
+        }
+
+        public function getLocale()
+        {
+            return 'en';
```
Comment (HeahDude) [inline reference/dic_tags.rst]: Following the comment above:
```suggestion
            return $this->locale;
```

File: reference/dic_tags.rst (hunk 1 above)
Comment (javiereguiluz) [reply reference/dic_tags.rst:459]: The merge branch is 4.4, but it's still OK to say that this was introduced in 4.3.

Comment (94noni) [reply reference/dic_tags.rst:459]: @javiereguiluz oki I thought it display only the oldest
also watch out I still see 4.3 in the destination branch

Capture d’écran 2022-10-26 à 12 08 41

File: reference/dic_tags.rst (hunk 2 above)
Comment (javiereguiluz) [reply reference/dic_tags.rst]: It doesn't add much, but all tags include a "purpose" line, so let's keep it.

**Comment by javiereguiluz** [comment]
Thank you Guillaume.

**Comment by javiereguiluz** [comment]
Merged in 4.4 and up. thanks to all reviewers! I added most of your suggestions.

**Pull request merged**